File: app/src/main/java/com/denchic45/kts/uivalidator/Rule.java
package com.denchic45.kts.uivalidator;

import java.util.function.BooleanSupplier;

public class Rule {
    private final BooleanSupplier validate;
    private final String errorMessage;
    private final int errorResId;

    public Rule(BooleanSupplier validate, String errorMessage) {
        this.validate = validate;
        this.errorMessage = errorMessage;
        errorResId = 0;
    }

    public Rule(BooleanSupplier validate, int errorResId) {
        this.validate = validate;
        this.errorResId = errorResId;
        errorMessage = null;
    }

    public Rule(BooleanSupplier validate) {
        this.validate = validate;
        this.errorResId = 0;
        errorMessage = null;
    }

    public boolean isValid() {
        return validate.getAsBoolean();
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public int getErrorResId() {
        return errorResId;
    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/Rule.kt
package com.denchic45.kts.uivalidator

import java.util.function.BooleanSupplier

class Rule {
    private val validate: BooleanSupplier
    val errorMessage: String?
    val errorResId: Int

    constructor(validate: BooleanSupplier, errorMessage: String?) {
        this.validate = validate
        this.errorMessage = errorMessage
        errorResId = 0
    }

    constructor(validate: BooleanSupplier, errorResId: Int) {
        this.validate = validate
        this.errorResId = errorResId
        errorMessage = null
    }

    constructor(validate: BooleanSupplier) {
        this.validate = validate
        errorResId = 0
        errorMessage = null
    }

    val isValid: Boolean
        get() = validate.asBoolean
}